test/test_helper_server.rb

```diff
@@ -0,0 +1,11 @@
+# typed: true
+# frozen_string_literal: true
+
+# Configure Rails Environment
+ENV["RAILS_ENV"] = "test"
+
+require "test_helper"
+
+require_relative "../test/dummy/config/environment"
```

Comment:
I think this is the key culprit here. I don't even think we need to switch fully to just Minitest.

As long as we require only the parts of Active Support related to testing, we can continue with the same setup.

But we can't load the dummy app for every test because that will load everything and get us into trouble.